lib/http2/hpack.c

```diff
@@ -240,7 +306,7 @@ static int decode_header(h2o_mem_pool_t *pool, struct st_h2o_decode_header_resul
                          const char **err_desc)
 {
     int32_t index = 0;
-    int value_is_indexed = 0, do_index = 0;
+    int value_is_indexed = 0, do_index = 0, ret = 0;
```

Comment:
In relation to https://github.com/h2o/h2o/pull/974/files#r71325566 I think there's no need to introduce `ret`. When reaching the bottom the function, the return code would either be `0` or `H2O_HTTP2_ERROR_INVALID_HEADER_CHAR` depending on if `*err_desc` is NULL.